source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -197,6 +213,43 @@ void JsonTranscoderConfig::addBuiltinSymbolDescriptor(const std::string& symbol_
   addFileDescriptor(file_proto);
 }
 
+Status JsonTranscoderConfig::createMethodInfo(const Protobuf::MethodDescriptor* descriptor,
+                                              const HttpRule& http_rule,
+                                              MethodInfoSharedPtr& method_info) {
+  method_info = std::make_shared<MethodInfo>();
+  method_info->descriptor_ = descriptor;
+  method_info->response_type_is_http_body_ =
+      descriptor->output_type()->full_name() == google::api::HttpBody::descriptor()->full_name();
+
+  const Protobuf::Type* request_type = type_helper_->Info()->GetTypeByTypeUrl(
+      Grpc::Common::typeUrl(descriptor->input_type()->full_name()));
+  if (request_type == nullptr) {
+    return ProtobufUtil::Status(Code::NOT_FOUND,
+                                "Could not resolve type: " + descriptor->input_type()->full_name());
+  }
+
+  if (http_rule.body() != "*") {
+    Status status = type_helper_->ResolveFieldPath(*request_type, http_rule.body(),
+                                                   &method_info->request_body_field_path);
+    if (!status.ok()) {
+      return status;
+    }
+
+    if (method_info->request_body_field_path.empty()) {
```

Comment:
do we handle case where request type is HttpBody and http_rule.body() is *?

Reply:
From reading google cloud's transcoding documentation it looks like `"*"` body is only applicable to json transcoding:
> The special name `*` can be used in the body mapping to define that every field not bound by the path template should be mapped to the request body.

https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#google.api.HttpRule

Reply:
Then,  do we support the HttpBody as the gRPC request type? 
with auto_mapping =true,   we generate a http rule with * body for every grpc method.

Reply:
Do you happen to know what is a behavior of Google Cloud in this case?
I'm probably inclined to treat `"*"` as `""` for `HttpBody` requests.